### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by seballgeyer**
Update of the pipeline to generate releases. 
tag extracted via the versionner package rather than manually, it allows us to install non tagged version.

### CODE REVIEW

**Comment by ronaldmaj** [comment]
We'll need to merge in https://github.com/GeoscienceAustralia/gnssanalysis/pull/32 first before this one right?

**Comment by seballgeyer** [comment]
> We'll need to merge in PR #32 first before this one right?

Shouldn't have any impact, but true that this one has the beginning of #32 in it, so to be on the safe side, yes.

**Review by ronaldmaj — COMMENTED** [review]
I think the `main` branch will now need to be merged back into this branch. I can see some changes from the PR #32 being reverted in this PR

File: .github/workflows/python-publish.yml
Comment (ronaldmaj) [inline .github/workflows/python-publish.yml:1]: If possible, this should be tested on GitHub to make sure the pipeline runs as expected. Not sure if that's possible before it's merged in

File: gnssanalysis/_version.py [hunk 1]
```diff
@@ -16,28 +16,35 @@
...
+    style: str
+    tag_prefix: str
+    parentdir_prefix: str
+    versionfile_source: str
+    verbose: bool
 
-def get_config():
+
+def get_config() -> VersioneerConfig:
     """Create, populate and return the VersioneerConfig() object."""
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:48]: One-line sphinx docstring (missing the description of what's returned, i.e. `:return`)

https://github.com/NilsJPWerner/autoDocstring/blob/master/docs/one-line-sphinx.md

File: gnssanalysis/_version.py [hunk 2]
```diff
@@ -59,9 +66,9 @@ class NotThisMethod(Exception):
 HANDLERS: Dict[str, Dict[str, Callable]] = {}
 
 
-def register_vcs_handler(vcs, method):  # decorator
+def register_vcs_handler(vcs: str, method: str) -> Callable:  # decorator
     """Create decorator to mark a method as the handler of a VCS."""
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:70]: One-line sphinx docstring (missing the description of input parameters, what's returned: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 3]
```diff
@@ -70,13 +77,19 @@ def decorate(f):
...
-                env=None):
+def run_command(
+    commands: List[str],
+    args: List[str],
+    cwd: Optional[str] = None,
+    verbose: bool = False,
+    hide_stderr: bool = False,
+    env: Optional[Dict[str, str]] = None,
+) -> Tuple[Optional[str], Optional[int]]:
     """Call the given command(s)."""
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:88]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 4]
```diff
@@ -113,7 +125,11 @@ def run_command(commands, args, cwd=None, verbose=False, hide_stderr=False,
     return stdout, process.returncode
 
 
-def versions_from_parentdir(parentdir_prefix, root, verbose):
+def versions_from_parentdir(
+    parentdir_prefix: str,
+    root: str,
+    verbose: bool,
+) -> Dict[str, Any]:
     """Try to determine the version from the parent directory name.
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:133]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 5]
```diff
@@ -138,13 +154,13 @@ def versions_from_parentdir(parentdir_prefix, root, verbose):
 
 
 @register_vcs_handler("git", "get_keywords")
-def git_get_keywords(versionfile_abs):
+def git_get_keywords(versionfile_abs: str) -> Dict[str, str]:
     """Extract version information from the given file."""
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:158]: One-line sphinx docstring (missing the description of input parameters, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 6]
```diff
@@ -166,7 +182,11 @@ def git_get_keywords(versionfile_abs):
 
 
 @register_vcs_handler("git", "keywords")
-def git_versions_from_keywords(keywords, tag_prefix, verbose):
+def git_versions_from_keywords(
+    keywords: Dict[str, str],
+    tag_prefix: str,
+    verbose: bool,
+) -> Dict[str, Any]:
     """Get version information from git keywords."""
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:190]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 7]
```diff
@@ -230,7 +250,12 @@ def git_versions_from_keywords(keywords, tag_prefix, verbose):
...
 
 @register_vcs_handler("git", "pieces_from_vcs")
-def git_pieces_from_vcs(tag_prefix, root, verbose, runner=run_command):
+def git_pieces_from_vcs(
+    tag_prefix: str,
+    root: str,
+    verbose: bool,
+    runner: Callable = run_command
+) -> Dict[str, Any]:
     """Get version from 'git describe' in the root of the source tree.
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:259]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 8]
```diff
@@ -362,14 +387,14 @@ def git_pieces_from_vcs(tag_prefix, root, verbose, runner=run_command):
     return pieces
 
 
-def plus_or_dot(pieces):
+def plus_or_dot(pieces: Dict[str, Any]) -> str:
     """Return a + if we don't already have one, else return a ."""
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:391]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 9]
```diff
@@ -362,14 +387,14 @@ def git_pieces_from_vcs(tag_prefix, root, verbose, runner=run_command):
...
     if "+" in pieces.get("closest-tag", ""):
         return "."
     return "+"
 
 
-def render_pep440(pieces):
+def render_pep440(pieces: Dict[str, Any]) -> str:
     """Build up version string, with post-release "local version identifier".
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:398]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 10]
```diff
@@ -394,7 +419,7 @@ def render_pep440(pieces):
     return rendered
 
 
-def render_pep440_branch(pieces):
+def render_pep440_branch(pieces: Dict[str, Any]) -> str:
     """TAG[[.dev0]+DISTANCE.gHEX[.dirty]] .
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:423]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 11]
```diff
@@ -424,7 +449,7 @@ def render_pep440_branch(pieces):
     return rendered
 
 
-def pep440_split_post(ver):
+def pep440_split_post(ver: str) -> Tuple[str, Optional[int]]:
     """Split pep440 version string at the post-release segment.
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:453]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 12]
```diff
@@ -434,7 +459,7 @@ def pep440_split_post(ver):
     return vc[0], int(vc[1] or 0) if len(vc) == 2 else None
 
 
-def render_pep440_pre(pieces):
+def render_pep440_pre(pieces: Dict[str, Any]) -> str:
     """TAG[.postN.devDISTANCE] -- No -dirty.
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:463]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 13]
```diff
@@ -458,7 +483,7 @@ def render_pep440_pre(pieces):
     return rendered
 
 
-def render_pep440_post(pieces):
+def render_pep440_post(pieces: Dict[str, Any]) -> str:
     """TAG[.postDISTANCE[.dev0]+gHEX] .
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:487]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 14]
```diff
@@ -485,7 +510,7 @@ def render_pep440_post(pieces):
     return rendered
 
 
-def render_pep440_post_branch(pieces):
+def render_pep440_post_branch(pieces: Dict[str, Any]) -> str:
     """TAG[.postDISTANCE[.dev0]+gHEX[.dirty]] .
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:514]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 15]
```diff
@@ -514,7 +539,7 @@ def render_pep440_post_branch(pieces):
     return rendered
 
 
-def render_pep440_old(pieces):
+def render_pep440_old(pieces: Dict[str, Any]) -> str:
     """TAG[.postDISTANCE[.dev0]] .
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:543]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 16]
```diff
@@ -536,7 +561,7 @@ def render_pep440_old(pieces):
     return rendered
 
 
-def render_git_describe(pieces):
+def render_git_describe(pieces: Dict[str, Any]) -> str:
     """TAG[-DISTANCE-gHEX][-dirty].
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:565]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 17]
```diff
@@ -556,7 +581,7 @@ def render_git_describe(pieces):
     return rendered
 
 
-def render_git_describe_long(pieces):
+def render_git_describe_long(pieces: Dict[str, Any]) -> str:
     """TAG-DISTANCE-gHEX[-dirty].
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:585]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 18]
```diff
@@ -576,7 +601,7 @@ def render_git_describe_long(pieces):
     return rendered
 
 
-def render(pieces, style):
+def render(pieces: Dict[str, Any], style: str) -> Dict[str, Any]:
     """Render the given version pieces into the requested style."""
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:605]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: gnssanalysis/_version.py [hunk 19]
```diff
@@ -612,7 +637,7 @@ def render(pieces, style):
             "date": pieces.get("date")}
 
 
-def get_versions():
+def get_versions() -> Dict[str, Any]:
     """Get version information or return default if unable to do so."""
```
Comment (ronaldmaj) [inline gnssanalysis/_version.py:641]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 20]
```diff
@@ -332,22 +333,40 @@
...
+    tag_prefix: str
+    versionfile_source: str
+    versionfile_build: Optional[str]
+    parentdir_prefix: Optional[str]
+    verbose: Optional[bool]
 
-def get_root():
+
+def get_root() -> str:
     """Get the project root directory.
```
Comment (ronaldmaj) [inline versioneer.py:346]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 21]
```diff
@@ -372,23 +391,24 @@ def get_root():
     return root
 
 
-def get_config_from_root(root):
+def get_config_from_root(root: str) -> VersioneerConfig:
     """Read the project setup.cfg file to determine Versioneer config."""
```
Comment (ronaldmaj) [inline versioneer.py:395]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 22]
```diff
@@ -419,22 +448,28 @@ class NotThisMethod(Exception):
 HANDLERS: Dict[str, Dict[str, Callable]] = {}
 
 
-def register_vcs_handler(vcs, method):  # decorator
+def register_vcs_handler(vcs: str, method: str) -> Callable:  # decorator
     """Create decorator to mark a method as the handler of a VCS."""
```
Comment (ronaldmaj) [inline versioneer.py:452]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 23]
```diff
@@ -419,22 +448,28 @@ class NotThisMethod(Exception):
...
-                env=None):
+def run_command(
+    commands: List[str],
+    args: List[str],
+    cwd: Optional[str] = None,
+    verbose: bool = False,
+    hide_stderr: bool = False,
+    env: Optional[Dict[str, str]] = None,
+) -> Tuple[Optional[str], Optional[int]]:
     """Call the given command(s)."""
```
Comment (ronaldmaj) [inline versioneer.py:468]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 24]
```diff
@@ -489,11 +523,11 @@ def run_command(commands, args, cwd=None, verbose=False, hide_stderr=False,
...
 import subprocess
 import sys
-from typing import Callable, Dict
+from typing import Any, Callable, Dict, List, Optional, Tuple
 import functools
 
 
-def get_keywords():
+def get_keywords() -> Dict[str, str]:
     """Get the keywords needed to look up the version information."""
```
Comment (ronaldmaj) [inline versioneer.py:531]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

- Also I've skipped repeating this message for this large block string, but from what I can see you can copy over from functions above down to here again? (The functions repeat from above)

File: versioneer.py [hunk 25]
```diff
@@ -1133,13 +1192,13 @@ def get_versions():
 
 
 @register_vcs_handler("git", "get_keywords")
-def git_get_keywords(versionfile_abs):
+def git_get_keywords(versionfile_abs: str) -> Dict[str, str]:
     """Extract version information from the given file."""
```
Comment (ronaldmaj) [inline versioneer.py:1196]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 26]
```diff
@@ -1161,7 +1220,11 @@ def git_get_keywords(versionfile_abs):
 
 
 @register_vcs_handler("git", "keywords")
-def git_versions_from_keywords(keywords, tag_prefix, verbose):
+def git_versions_from_keywords(
+    keywords: Dict[str, str],
+    tag_prefix: str,
+    verbose: bool,
+) -> Dict[str, Any]:
     """Get version information from git keywords."""
```
Comment (ronaldmaj) [inline versioneer.py:1228]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 27]
```diff
@@ -1225,7 +1288,12 @@ def git_versions_from_keywords(keywords, tag_prefix, verbose):
...
 
 @register_vcs_handler("git", "pieces_from_vcs")
-def git_pieces_from_vcs(tag_prefix, root, verbose, runner=run_command):
+def git_pieces_from_vcs(
+    tag_prefix: str,
+    root: str,
+    verbose: bool,
+    runner: Callable = run_command
+) -> Dict[str, Any]:
     """Get version from 'git describe' in the root of the source tree.
```
Comment (ronaldmaj) [inline versioneer.py:1297]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 28]
```diff
@@ -1357,7 +1425,7 @@ def git_pieces_from_vcs(tag_prefix, root, verbose, runner=run_command):
     return pieces
 
 
-def do_vcs_install(versionfile_source, ipy):
+def do_vcs_install(versionfile_source: str, ipy: Optional[str]) -> None:
     """Git-specific installation logic for Versioneer.
```
Comment (ronaldmaj) [inline versioneer.py:1429]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 29]
```diff
@@ -1395,7 +1463,11 @@ def do_vcs_install(versionfile_source, ipy):
     run_command(GITS, ["add", "--"] + files)
 
 
-def versions_from_parentdir(parentdir_prefix, root, verbose):
+def versions_from_parentdir(
+    parentdir_prefix: str,
+    root: str,
+    verbose: bool,
+) -> Dict[str, Any]:
     """Try to determine the version from the parent directory name.
```
Comment (ronaldmaj) [inline versioneer.py:1471]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 30]
```diff
@@ -1437,7 +1509,7 @@ def get_versions():
 """
 
 
-def versions_from_file(filename):
+def versions_from_file(filename: str) -> Dict[str, Any]:
     """Try to determine the version from _version.py if present."""
```
Comment (ronaldmaj) [inline versioneer.py:1513]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 31]
```diff
@@ -1454,9 +1526,8 @@ def versions_from_file(filename):
     return json.loads(mo.group(1))
 
 
-def write_to_version_file(filename, versions):
+def write_to_version_file(filename: str, versions: Dict[str, Any]) -> None:
     """Write the given version number to the given _version.py file."""
```
Comment (ronaldmaj) [inline versioneer.py:1530]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 32]
```diff
@@ -1465,14 +1536,14 @@ def write_to_version_file(filename, versions):
     print("set %s to '%s'" % (filename, versions["version"]))
 
 
-def plus_or_dot(pieces):
+def plus_or_dot(pieces: Dict[str, Any]) -> str:
     """Return a + if we don't already have one, else return a ."""
```
Comment (ronaldmaj) [inline versioneer.py:1540]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 33]
```diff
@@ -1465,14 +1536,14 @@ def write_to_version_file(filename, versions):
...
     if "+" in pieces.get("closest-tag", ""):
         return "."
     return "+"
 
 
-def render_pep440(pieces):
+def render_pep440(pieces: Dict[str, Any]) -> str:
     """Build up version string, with post-release "local version identifier".
```
Comment (ronaldmaj) [inline versioneer.py:1547]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 34]
```diff
@@ -1497,7 +1568,7 @@ def render_pep440(pieces):
     return rendered
 
 
-def render_pep440_branch(pieces):
+def render_pep440_branch(pieces: Dict[str, Any]) -> str:
     """TAG[[.dev0]+DISTANCE.gHEX[.dirty]] .
```
Comment (ronaldmaj) [inline versioneer.py:1572]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 35]
```diff
@@ -1527,7 +1598,7 @@ def render_pep440_branch(pieces):
     return rendered
 
 
-def pep440_split_post(ver):
+def pep440_split_post(ver: str) -> Tuple[str, Optional[int]]:
     """Split pep440 version string at the post-release segment.
```
Comment (ronaldmaj) [inline versioneer.py:1602]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 36]
```diff
@@ -1537,7 +1608,7 @@ def pep440_split_post(ver):
     return vc[0], int(vc[1] or 0) if len(vc) == 2 else None
 
 
-def render_pep440_pre(pieces):
+def render_pep440_pre(pieces: Dict[str, Any]) -> str:
     """TAG[.postN.devDISTANCE] -- No -dirty.
```
Comment (ronaldmaj) [inline versioneer.py:1612]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 37]
```diff
@@ -1561,7 +1632,7 @@ def render_pep440_pre(pieces):
     return rendered
 
 
-def render_pep440_post(pieces):
+def render_pep440_post(pieces: Dict[str, Any]) -> str:
     """TAG[.postDISTANCE[.dev0]+gHEX] .
```
Comment (ronaldmaj) [inline versioneer.py:1636]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 38]
```diff
@@ -1588,7 +1659,7 @@ def render_pep440_post(pieces):
     return rendered
 
 
-def render_pep440_post_branch(pieces):
+def render_pep440_post_branch(pieces: Dict[str, Any]) -> str:
     """TAG[.postDISTANCE[.dev0]+gHEX[.dirty]] .
```
Comment (ronaldmaj) [inline versioneer.py:1663]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 39]
```diff
@@ -1617,7 +1688,7 @@ def render_pep440_post_branch(pieces):
     return rendered
 
 
-def render_pep440_old(pieces):
+def render_pep440_old(pieces: Dict[str, Any]) -> str:
     """TAG[.postDISTANCE[.dev0]] .
```
Comment (ronaldmaj) [inline versioneer.py:1692]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 40]
```diff
@@ -1639,7 +1710,7 @@ def render_pep440_old(pieces):
     return rendered
 
 
-def render_git_describe(pieces):
+def render_git_describe(pieces: Dict[str, Any]) -> str:
     """TAG[-DISTANCE-gHEX][-dirty].
```
Comment (ronaldmaj) [inline versioneer.py:1714]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 41]
```diff
@@ -1659,7 +1730,7 @@ def render_git_describe(pieces):
     return rendered
 
 
-def render_git_describe_long(pieces):
+def render_git_describe_long(pieces: Dict[str, Any]) -> str:
     """TAG-DISTANCE-gHEX[-dirty].
```
Comment (ronaldmaj) [inline versioneer.py:1734]: Might be good to start with a description and then this
Also, One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 42]
```diff
@@ -1679,7 +1750,7 @@ def render_git_describe_long(pieces):
     return rendered
 
 
-def render(pieces, style):
+def render(pieces: Dict[str, Any], style: str) -> Dict[str, Any]:
     """Render the given version pieces into the requested style."""
```
Comment (ronaldmaj) [inline versioneer.py:1754]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: `:param` `:return`)

File: versioneer.py [hunk 43]
```diff
@@ -1719,7 +1790,7 @@ class VersioneerBadRootError(Exception):
     """The project root directory is unknown or missing key files."""
 
 
-def get_versions(verbose=False):
+def get_versions(verbose: bool = False) -> Dict[str, Any]:
     """Get the project version from whatever source is available.
```
Comment (ronaldmaj) [inline versioneer.py:1794]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: :param :return)

File: versioneer.py [hunk 44]
```diff
@@ -1795,12 +1866,12 @@ def get_versions(verbose=False):
             "date": None}
 
 
-def get_version():
+def get_version() -> str:
     """Get the short version string for this project."""
```
Comment (ronaldmaj) [inline versioneer.py:1870]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: :param :return)

File: versioneer.py [hunk 45]
```diff
@@ -1795,12 +1866,12 @@ def get_versions(verbose=False):
...
     return get_versions()["version"]
 
 
-def get_cmdclass(cmdclass=None):
+def get_cmdclass(cmdclass: Optional[Dict[str, Any]] = None):
     """Get the custom setuptools subclasses used by Versioneer.
```
Comment (ronaldmaj) [inline versioneer.py:1875]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: :param :return)

File: versioneer.py [hunk 46]
```diff
@@ -2099,7 +2170,7 @@ def make_release_tree(self, base_dir, files):
 """
 
 
-def do_setup():
+def do_setup() -> int:
     """Do main VCS-independent setup function for installing Versioneer."""
```
Comment (ronaldmaj) [inline versioneer.py:2174]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: :param :return)

File: versioneer.py [hunk 47]
```diff
@@ -2146,16 +2218,16 @@ def do_setup():
...
     # .gitattributes to mark _version.py for export-subst keyword
     # substitution.
-    do_vcs_install(cfg.versionfile_source, ipy)
+    do_vcs_install(cfg.versionfile_source, maybe_ipy)
     return 0
 
 
-def scan_setup_py():
+def scan_setup_py() -> int:
     """Validate the contents of setup.py against Versioneer's expectations."""
```
Comment (ronaldmaj) [inline versioneer.py:2231]: One-line sphinx docstring (missing the description of input parameters, errors raised, what's returned, etc.: :param :return)

**Review by ronaldmaj — CHANGES_REQUESTED** [review]
Really good stuff, I think this will make it easier to develop and maintain the repo moving forward.
And sorry for so many repeat comments, but just wanted to make it easier to make sure all the docstrings are hit when you are running through it.

File: gnssanalysis/_version.py (hunk 2 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:70]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 4 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:133]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 5 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:158]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 6 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:190]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 7 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:259]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 8 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:391]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 10 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:423]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 12 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:463]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 13 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:487]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 14 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:514]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 16 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:565]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 17 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:585]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 18 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:605]: file autogenerated by the versionner script. so won't be addressed

File: gnssanalysis/_version.py (hunk 19 above)
Comment (seballgeyer) [reply gnssanalysis/_version.py:641]: file autogenerated by the versionner script. so won't be addressed

File: .github/workflows/python-publish.yml
Comment (seballgeyer) [reply .github/workflows/python-publish.yml:1]: as far as I can test (without triggereing the pipeline), it seems to work. 
i.e. get the proper tag on the package generated.

**Review by ronaldmaj — APPROVED** [review]
Comments have been addressed, happy for this to go in

**Pull request merged**